In [72]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [73]:
# Importing ML Libraries
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

In [74]:
# Importing the dataset
df = pd.read_csv("C:/Users/Tushar Goel/Desktop/Intern related dataset/gold.csv ")

In [75]:
df.head()

Date  Price   Open   High    Low    Vol. Change %   Pred  \
0  May 04, 2017  28060  28400  28482  28025   0.08K   -1.79%  738.0   
1  May 05, 2017  28184  28136  28382  28135   0.06K    0.44% -146.0   
2  May 08, 2017  28119  28145  28255  28097   7.85K   -0.23%   30.0   
3  May 09, 2017  27981  28125  28192  27947  10.10K   -0.49%  357.0   
4  May 10, 2017  28007  28060  28146  27981   9.28K    0.09%  124.0   

          new  
0  117.570740  
1  295.430176  
2  132.123714  
3  101.298064  
4  112.153318

In [76]:
# Calculating no. of data fields which are empty in Pred
df['Pred'].isnull().value_counts()

False    411
True     101
Name: Pred, dtype: int64

In [77]:
# Calculating no. of data fields which are empty in New
df['new'].isnull().value_counts()

False    512
Name: new, dtype: int64

In [78]:
df.corr()

Price      Open      High       Low      Pred       new
Price  1.000000  0.989758  0.995125  0.996830  0.040489  0.164504
Open   0.989758  1.000000  0.995988  0.993992  0.109153  0.058551
High   0.995125  0.995988  1.000000  0.994194  0.080394  0.142109
Low    0.996830  0.993992  0.994194  1.000000  0.014411  0.091895
Pred   0.040489  0.109153  0.080394  0.014411  1.000000 -0.006109
new    0.164504  0.058551  0.142109  0.091895 -0.006109  1.000000

## as new is highly coorelated with OHLC , it is highly possible that it is linear related to them.
## Lets check it out by training our model


In [79]:
y_1 = df['Pred'].loc[:410]
x_2 = df.iloc[:,[1,2,3,4]]
x_1 = x_2.loc[:410]
y_2 = df['new']

In [80]:
y_1.isnull().value_counts()

False    411
Name: Pred, dtype: int64

## Now we are going to train our model

### First we going to determine which column is polynomial related or linear related to the inputs by comparing the accuracies (r2_score)
### we are goin to prepare our model for 'New' column

In [81]:
from sklearn.preprocessing import PolynomialFeatures

In [82]:
# spliting the data
x_2_train,x_2_test,y_2_train,y_2_test = train_test_split(x_2,y_2,test_size=0.1,random_state=0)

In [83]:
lr = LinearRegression()

In [84]:
linear_model = lr.fit(x_2_train,y_2_train)

In [85]:
y_2_predict = linear_model.predict(x_2_test)

In [86]:
print(r2_score(y_2_test,y_2_predict))

0.9999837212433824


## It is 99.99 % accurate . Hence "New" is Linear related
## Then Pred is Polynomially related

## Let's Train our model for Pred

In [87]:
poly = PolynomialFeatures(3)

x_1 = poly.fit_transform(x_1)

In [88]:
x_1_train,x_1_test,y_1_train,y_1_test = train_test_split(x_1,y_1,test_size=0.1,random_state=0)

In [89]:
lr_1 = LinearRegression()
linear_model_1 = lr_1.fit(x_1_train,y_1_train)

In [90]:
y_1_predict = linear_model_1.predict(x_1_test)

In [91]:
print(r2_score(y_1_test,y_1_predict))

0.9999999706245378


In [244]:
linear_model_1.coef_

array([-2.30044937e+00,  1.69535261e+00,  3.34348448e+00, -1.21476771e+00,
       -3.81914969e+00, -2.15493329e-06, -5.45707096e-07,  1.31649546e-05,
        1.21238936e-05, -3.01261181e-05,  8.73733377e-06,  2.92416079e-05,
       -1.97681806e-05,  3.22417839e-05, -4.31325302e-05,  2.92574912e-08,
       -4.02505344e-10, -3.40598575e-08, -5.33685578e-08, -5.86635140e-09,
        9.11031510e-09,  3.40810219e-09,  9.76745415e-10,  5.68528902e-08,
        2.30650036e-08, -1.43594306e-08,  2.75853614e-08,  2.23520590e-08,
       -1.92002701e-08, -2.60808126e-08, -1.15203368e-08,  5.55312042e-09,
        2.13428158e-09, -1.81366217e-08,  2.70241050e-09])

## This Polynomial model is 99.99% accurate (Pred Column)
## Let's fill our columns fields by this model

In [99]:
value = df.iloc[:,[1,2,3,4]].loc[411:]

In [101]:
value = poly.fit_transform(value)

In [107]:
a = linear_model_1.predict(value)

In [108]:
a

array([ 852.01634807,  422.1353249 ,  530.08192832,  144.04961614,
        414.92184796, 1103.26379864,  411.03271235,  612.95374625,
          5.00650984,  546.96030187,  -27.00114288,  367.92368908,
        263.07609931,  614.94515286,  478.98243822,  -80.76077507,
        489.09802404, 2892.15812884,  339.8814995 ,  273.92997871,
        468.68405176,  487.24279144,  -47.94418991,  363.89278406,
          6.92604296,  276.86042553,  283.16935774,  550.32302238,
        110.06576739,  412.82646216,  146.07691107,  241.19149827,
        405.52794876,  118.81021948,  403.99355898,  303.16862693,
        153.3677738 ,  677.45927816, 2115.83035946,  860.67955258,
        -36.90370442,  615.42607091,  467.66059022,  716.40454696,
        410.52379975,  250.37257739,  465.72742311,  253.11532157,
        108.83135643,  468.62273928,  241.65852415,  777.1890483 ,
        470.84113652,  555.86085792, -138.93941107,  224.31179381,
        603.74041924, 3023.19029605,  330.72853665, 2898.74004

In [110]:
df['Pred'].loc[411:] = a

In [111]:
df['Pred']

0       738.000000
1      -146.000000
2        30.000000
3       357.000000
4       124.000000
5       149.000000
6       167.000000
7        22.000000
8        34.000000
9       420.000000
10      234.000000
11      345.000000
12      265.000000
13      138.000000
14      378.000000
15      413.000000
16      214.000000
17      121.000000
18      491.000000
19      397.000000
20      404.000000
21      222.000000
22       -3.000000
23      259.000000
24      344.000000
25      906.000000
26      526.000000
27      174.000000
28      685.000000
29      329.000000
          ...     
482      93.039262
483     740.585250
484     466.036095
485     502.197352
486     159.163982
487     833.320565
488     377.887136
489       6.040264
490     693.994548
491     518.016466
492     553.994364
493     516.972408
494     201.659231
495     338.136910
496     488.859425
497    1070.469054
498      -7.963816
499     578.952531
500     468.114854
501     223.088379
502     264.053517
503    -131.

# We are able to fill our columns through our model

The S&P 500 is a stock market index that tracks the stocks of 500 large-cap U.S. companies. It represents the stock market's performance by reporting the risks and returns of the biggest companies. Investors use it as the benchmark of the overall market, to which all other investments are compared.

### Importing Stock and Nifty Index

In [162]:
nifty = pd.read_csv("C:/Users/Tushar Goel/Desktop/Intern related dataset/Nifty50.csv ")
stock = pd.read_csv("C:/Users/Tushar Goel/Desktop/Intern related dataset/Small_Cap/ASHOKA.csv ")

In [163]:
nifty.head()

Date     Open     High      Low    Close  Shares Traded  \
0  15-May-2017  9433.55  9449.25  9423.10  9445.40      145163953   
1  16-May-2017  9461.00  9517.20  9456.35  9512.25      171709433   
2  17-May-2017  9517.60  9532.60  9486.10  9525.75      211856040   
3  18-May-2017  9453.20  9489.10  9418.10  9429.45      199340647   
4  19-May-2017  9469.90  9505.75  9390.75  9427.90      259861396   

   Turnover (Rs. Cr)  
0            7790.97  
1            9209.89  
2           10726.23  
3           10782.46  
4           11544.77

In [164]:
stock.head()

Symbol Series         Date  Prev Close  Open Price  High Price  Low Price  \
0  ASHOKA     EQ  15-May-2017      210.35       211.0       217.5     206.15   
1  ASHOKA     EQ  16-May-2017      216.05       212.3       216.0     211.00   
2  ASHOKA     EQ  17-May-2017      214.90       214.3       218.7     212.10   
3  ASHOKA     EQ  18-May-2017      217.00       216.0       216.6     208.00   
4  ASHOKA     EQ  19-May-2017      209.65       212.1       213.9     205.05   

   Last Price  Close Price  Average Price  Total Traded Quantity     Turnover  \
0      216.30       216.05         212.63                 159997  34020520.45   
1      214.00       214.90         213.24                 213214  45466299.10   
2      216.55       217.00         215.97                 383131  82743402.10   
3      210.00       209.65         211.74                  78709  16665707.55   
4      206.05       206.25         209.95                 398725  83713500.05   

   No. of Trades  Deliverable Qty  % Dly Qt to Traded Qty  
0           3506            88509                   55.32  
1           2860           151941                   71.26  
2           3297           261939                   68.37  
3           1372            35458                   45.05  
4           2060           301747                   75.68

## Beta Value on Daily Basis

### STOCK RETURN

In [165]:
# Calculating the Daily Return :
stock['Daily_Return'] = stock['Close Price'].pct_change()

In [166]:
stock.head()

Symbol Series         Date  Prev Close  Open Price  High Price  Low Price  \
0  ASHOKA     EQ  15-May-2017      210.35       211.0       217.5     206.15   
1  ASHOKA     EQ  16-May-2017      216.05       212.3       216.0     211.00   
2  ASHOKA     EQ  17-May-2017      214.90       214.3       218.7     212.10   
3  ASHOKA     EQ  18-May-2017      217.00       216.0       216.6     208.00   
4  ASHOKA     EQ  19-May-2017      209.65       212.1       213.9     205.05   

   Last Price  Close Price  Average Price  Total Traded Quantity     Turnover  \
0      216.30       216.05         212.63                 159997  34020520.45   
1      214.00       214.90         213.24                 213214  45466299.10   
2      216.55       217.00         215.97                 383131  82743402.10   
3      210.00       209.65         211.74                  78709  16665707.55   
4      206.05       206.25         209.95                 398725  83713500.05   

   No. of Trades  Deliverable Qty  % Dly Qt to Traded Qty  Daily_Return  
0           3506            88509                   55.32           NaN  
1           2860           151941                   71.26     -0.005323  
2           3297           261939                   68.37      0.009772  
3           1372            35458                   45.05     -0.033871  
4           2060           301747                   75.68     -0.016218

### MARKET RETURN

In [167]:
nifty['Daily_Return'] = nifty['Close'].pct_change()

In [168]:
nifty.head()

Date     Open     High      Low    Close  Shares Traded  \
0  15-May-2017  9433.55  9449.25  9423.10  9445.40      145163953   
1  16-May-2017  9461.00  9517.20  9456.35  9512.25      171709433   
2  17-May-2017  9517.60  9532.60  9486.10  9525.75      211856040   
3  18-May-2017  9453.20  9489.10  9418.10  9429.45      199340647   
4  19-May-2017  9469.90  9505.75  9390.75  9427.90      259861396   

   Turnover (Rs. Cr)  Daily_Return  
0            7790.97           NaN  
1            9209.89      0.007078  
2           10726.23      0.001419  
3           10782.46     -0.010109  
4           11544.77     -0.000164

In [169]:
# Converting both datasets into DateTimeIndex
stock['Date'] = stock['Date'].astype('datetime64[ns]')
nifty['Date'] = nifty['Date'].astype('datetime64[ns]')
stock = stock.set_index('Date')
nifty = nifty.set_index('Date')

In [170]:
ts = stock.last('90D')


Symbol Series  Prev Close  Open Price  High Price  Low Price  \
Date                                                                       
2019-02-13  ASHOKA     EQ      111.10      110.00      114.05     110.00   
2019-02-14  ASHOKA     EQ      110.80      110.80      112.60     110.55   
2019-02-15  ASHOKA     EQ      111.80      112.50      114.90     110.00   
2019-02-18  ASHOKA     EQ      112.25      113.00      114.40     107.25   
2019-02-19  ASHOKA     EQ      109.80      109.60      112.25     106.70   
2019-02-20  ASHOKA     EQ      109.10      108.55      112.90     107.65   
2019-02-21  ASHOKA     EQ      111.95      111.10      124.25     109.80   
2019-02-22  ASHOKA     EQ      121.20      121.40      125.40     120.20   
2019-02-25  ASHOKA     EQ      123.45      124.00      124.00     117.50   
2019-02-26  ASHOKA     EQ      119.05      117.75      117.75     112.60   
2019-02-27  ASHOKA     EQ      116.40      118.60      120.20     114.00   
2019-02-28  ASHOKA     EQ      114.45      113.20      119.70     113.20   
2019-03-01  ASHOKA     EQ      118.50      120.00      122.40     118.60   
2019-03-05  ASHOKA     EQ      121.20      121.30      132.00     121.00   
2019-03-06  ASHOKA     EQ      129.55      130.80      141.70     130.15   
2019-03-07  ASHOKA     EQ      137.55      138.40      138.70     133.05   
2019-03-08  ASHOKA     EQ      134.45      132.25      136.35     131.50   
2019-03-11  ASHOKA     EQ      133.95      134.90      139.00     133.85   
2019-03-12  ASHOKA     EQ      137.35      139.40      150.00     138.95   
2019-03-13  ASHOKA     EQ      141.20      141.25      142.00     136.00   
2019-03-14  ASHOKA     EQ      140.15      139.35      141.55     136.85   
2019-03-15  ASHOKA     EQ      139.30      138.15      140.20     136.00   
2019-03-18  ASHOKA     EQ      137.45      137.50      139.90     135.95   
2019-03-19  ASHOKA     EQ      137.85      138.50      139.50     133.55   
2019-03-20  ASHOKA     EQ      134.05      134.30      135.45     128.00   
2019-03-22  ASHOKA     EQ      128.85      129.00      132.00     127.50   
2019-03-25  ASHOKA     EQ      129.15      128.00      128.50     123.05   
2019-03-26  ASHOKA     EQ      123.95      124.55      126.15     124.55   
2019-03-27  ASHOKA     EQ      125.85      127.00      131.65     124.55   
2019-03-28  ASHOKA     EQ      130.95      132.00      132.90     128.55   
2019-03-29  ASHOKA     EQ      129.80      131.80      132.50     126.25   
2019-04-01  ASHOKA     EQ      126.90      128.30      131.00     127.50   
2019-04-02  ASHOKA     EQ      130.20      130.95      135.25     130.35   
2019-04-03  ASHOKA     EQ      133.00      132.00      135.90     132.00   
2019-04-04  ASHOKA     EQ      133.10      132.00      132.90     126.00   
2019-04-05  ASHOKA     EQ      127.80      128.70      137.00     128.10   
2019-04-08  ASHOKA     EQ      134.45      134.50      136.70     131.10   
2019-04-09  ASHOKA     EQ      132.85      132.80      133.60     130.00   
2019-04-10  ASHOKA     EQ      131.40      130.50      132.50     129.00   
2019-04-11  ASHOKA     EQ      129.35      129.90      131.50     128.40   
2019-04-12  ASHOKA     EQ      129.25      130.00      131.55     129.50   
2019-04-15  ASHOKA     EQ      130.90      131.80      132.00     129.50   
2019-04-16  ASHOKA     EQ      130.00      130.00      133.30     130.00   
2019-04-18  ASHOKA     EQ      131.10      134.80      135.00     130.20   
2019-04-22  ASHOKA     EQ      131.65      132.55      132.55     127.10   
2019-04-23  ASHOKA     EQ      128.05      128.00      129.60     124.50   
2019-04-24  ASHOKA     EQ      127.85      129.00      130.45     128.20   
2019-04-25  ASHOKA     EQ      129.80      130.90      130.90     128.55   
2019-04-26  ASHOKA     EQ      129.05      129.05      129.80     127.05   
2019-04-30  ASHOKA     EQ      127.60      126.15      127.45     123.50   
2019-05-02  ASHOKA     EQ      124.20      124.20      124.2

In [179]:
y_daily = ts.iloc[:,-1].values

In [183]:
# Stock return
y_daily = y_daily.reshape(-1,1)

In [173]:
tn = nifty.last('90D')


In [184]:
# MArket return
x_daily = tn.iloc[:,-1].values
x_daily = x_daily.reshape(-1,1)

In [185]:
daily = LinearRegression()
beta_daily = daily.fit(x_daily,y_daily)

### Beta Value for Daily 

In [186]:
beta_daily.coef_

array([[1.60861623]])

Therefore Beta Value on daily basis is 1.60861623

# Beta Value for Monthly Return

## Monthly Close price for Nifty and Stock

In [191]:
stock.reset_index(inplace=True)

In [193]:
stock['month'] = pd.DatetimeIndex(stock['Date']).month

In [195]:
nifty.reset_index(inplace=True)

In [196]:
nifty['month'] = pd.DatetimeIndex(nifty['Date']).month

In [197]:
stock.head()

Date  Symbol Series  Prev Close  Open Price  High Price  Low Price  \
0 2017-05-15  ASHOKA     EQ      210.35       211.0       217.5     206.15   
1 2017-05-16  ASHOKA     EQ      216.05       212.3       216.0     211.00   
2 2017-05-17  ASHOKA     EQ      214.90       214.3       218.7     212.10   
3 2017-05-18  ASHOKA     EQ      217.00       216.0       216.6     208.00   
4 2017-05-19  ASHOKA     EQ      209.65       212.1       213.9     205.05   

   Last Price  Close Price  Average Price  Total Traded Quantity     Turnover  \
0      216.30       216.05         212.63                 159997  34020520.45   
1      214.00       214.90         213.24                 213214  45466299.10   
2      216.55       217.00         215.97                 383131  82743402.10   
3      210.00       209.65         211.74                  78709  16665707.55   
4      206.05       206.25         209.95                 398725  83713500.05   

   No. of Trades  Deliverable Qty  % Dly Qt to Traded Qty  Daily_Return  month  
0           3506            88509                   55.32           NaN      5  
1           2860           151941                   71.26     -0.005323      5  
2           3297           261939                   68.37      0.009772      5  
3           1372            35458                   45.05     -0.033871      5  
4           2060           301747                   75.68     -0.016218      5

In [198]:
stock_month = stock.groupby('month')

In [200]:
stock_month.first()

Date  Symbol Series  Prev Close  Open Price  High Price  \
month                                                                 
1     2018-01-01  ASHOKA     EQ      244.65      245.00      250.80   
2     2018-02-01  ASHOKA     EQ      227.00      232.00      233.95   
3     2018-03-01  ASHOKA     EQ      219.00      226.00      235.60   
4     2018-04-02  ASHOKA     EQ      248.45      250.00      264.50   
5     2017-05-15  ASHOKA     EQ      210.35      211.00      217.50   
6     2017-06-01  ASHOKA     EQ      190.35      191.10      192.90   
7     2017-07-03  ASHOKA     EQ      184.60      183.55      187.55   
8     2017-08-01  ASHOKA     EQ      185.55      185.80      185.80   
9     2017-09-01  ASHOKA     EQ      195.45      197.20      197.20   
10    2017-10-03  ASHOKA     EQ      187.70      187.20      190.00   
11    2017-11-01  ASHOKA     EQ      212.95      213.00      219.00   
12    2017-12-01  ASHOKA     EQ      251.80      254.95      258.85   

       Low Price  Last Price  Close Price  Average Price  \
month                                                      
1         242.05      245.50       245.95         247.76   
2         226.10      232.10       230.00         230.28   
3         222.50      225.75       225.35         230.23   
4         250.00      256.00       256.15         259.20   
5         206.15      216.30       216.05         212.63   
6         186.15      187.75       187.50         189.06   
7         183.55      185.90       185.75         185.52   
8         179.05      179.05       179.95         181.25   
9         191.70      193.75       192.75         193.86   
10        182.15      182.85       183.30         184.13   
11        203.50      218.00       216.70         214.62   
12        240.00      241.10       243.80         251.03   

       Total Traded Quantity      Turnover  No. of Trades  Deliverable Qty  \
month                                                                        
1                     203618  5.044898e+07           2297           136264   
2                     500813  1.153250e+08          13064           300979   
3                    1047709  2.412153e+08          19349           275038   
4                     418566  1.084937e+08           8185           188070   
5                     159997  3.402052e+07           3506            88509   
6                      63366  1.198017e+07           1427            39893   
7                      30493  5.657131e+06            844            17948   
8                      86648  1.570463e+07           1283            67086   
9                     101583  1.969290e+07           2065            52231   
10                    499796  9.202611e+07           1747           459812   
11                   1211278  2.599672e+08          20826           737367   
12                    256827  6.447191e+07          10480           131030   

       % Dly Qt to Traded Qty  Daily_Return  
month                                        
1                       66.92      0.005314  
2                       60.10      0.013216  
3                       26.25      0.028995  
4                       44.93      0.030992  
5                       55.32     -0.005323  
6                       62.96     -0.014972  
7                       58.86      0.006230  
8                       77.42     -0.030181  
9                       51.42     -0.013814  
10                      92.00     -0.023442  
11                      60.88      0.017610  
12                      51.02     -0.031771

In [210]:
monthly_stock_close_price = list()
for name,group in stock_month:
        monthly_stock_close_price.append(group['Close Price'].sum())

In [230]:
monthly_return = pd.DataFrame()

In [231]:
monthly_return['Monthly_Stock_close_price'] = monthly_stock_close_price

In [222]:
nifty_month = nifty.groupby('month')

In [223]:
nifty_month.first()

Date      Open      High       Low     Close  Shares Traded  \
month                                                                     
1     2018-01-01  10531.70  10537.85  10423.10  10435.55      134532090   
2     2018-02-01  11044.55  11117.35  10878.80  11016.90      315743486   
3     2018-03-01  10479.95  10525.50  10447.15  10458.35      181378504   
4     2018-04-02  10151.65  10220.10  10127.75  10211.80      223135459   
5     2017-05-15   9433.55   9449.25   9423.10   9445.40      145163953   
6     2017-06-01   9603.55   9634.65   9589.90   9616.10      181533283   
7     2017-07-03   9587.95   9624.00   9543.55   9615.00      188690599   
8     2017-08-01  10101.05  10128.60  10065.75  10114.65      190000516   
9     2017-09-01   9937.65   9983.45   9909.85   9974.40      162730656   
10    2017-10-03   9893.30   9895.40   9831.05   9859.50      167833406   
11    2017-11-01  10390.35  10451.65  10383.05  10440.50      268270466   
12    2017-12-01  10263.70  10272.70  10108.55  10121.80      147765528   

       Turnover (Rs. Cr)  Daily_Return  
month                                   
1                7546.56     -0.009035  
2               17719.40     -0.000979  
3               10179.48     -0.003288  
4               10968.95      0.009700  
5                7790.97      0.007078  
6                9482.61     -0.000535  
7                8720.76      0.009884  
8               11515.29      0.003726  
9                9314.82      0.005697  
10               9698.55      0.007243  
11              14206.21      0.010179  
12               8518.58     -0.010243

In [224]:
monthly_market_close_price = list()
for name,group in nifty_month:
        monthly_market_close_price.append(group['Close'].sum())

In [232]:
monthly_return['Monthly_market_close_price'] = monthly_market_close_price

In [233]:
# let us see our monthly close price of stock and market
monthly_return

Monthly_Stock_close_price  Monthly_market_close_price
0                     8173.40                   485583.05
1                     6606.05                   416805.80
2                     6941.70                   409447.35
3                     8074.55                   441594.20
4                     9422.40                   449720.60
5                     8834.25                   427348.50
6                     7915.05                   448657.95
7                     6868.60                   449392.10
8                     6241.35                   412883.30
9                     6210.45                   420833.55
10                    7487.45                   439580.45
11                    7430.20                   422014.10

In [234]:
monthly_return['monthly_stock_return'] = monthly_return['Monthly_Stock_close_price'].pct_change()

In [235]:
monthly_return['monthly_market_return'] = monthly_return['Monthly_market_close_price'].pct_change()

In [236]:
monthly_return

Monthly_Stock_close_price  Monthly_market_close_price  \
0                     8173.40                   485583.05   
1                     6606.05                   416805.80   
2                     6941.70                   409447.35   
3                     8074.55                   441594.20   
4                     9422.40                   449720.60   
5                     8834.25                   427348.50   
6                     7915.05                   448657.95   
7                     6868.60                   449392.10   
8                     6241.35                   412883.30   
9                     6210.45                   420833.55   
10                    7487.45                   439580.45   
11                    7430.20                   422014.10   

    monthly_stock_return  monthly_market_return  
0                    NaN                    NaN  
1              -0.191762              -0.141638  
2               0.050809              -0.017654  
3               0.163195               0.078513  
4               0.166926               0.018402  
5              -0.062420              -0.049747  
6              -0.104050               0.049864  
7              -0.132210               0.001636  
8              -0.091321              -0.081240  
9              -0.004951               0.019255  
10              0.205621               0.044547  
11             -0.007646              -0.039962

In [240]:
monthly_return.drop(0,inplace=True)

# lets train our model for monthly stock and market return

In [241]:
x_monthly = monthly_return['monthly_market_return'].values.reshape(-1,1)
y_monthly = monthly_return['monthly_stock_return'].values.reshape(-1,1)

In [242]:
lr = LinearRegression()
linear_model_monthly = lr.fit(x_monthly,y_monthly)

In [243]:
linear_model_monthly.coef_

array([[1.37515766]])

Therefore Beta Value for monthly return is 1.37515766

# Beta Value for Daily return is 1.60861623
# Beta Value for Monthly Return is 1.37515766

The Beta Value for both Daily return and Monthly return are more than 1 .

Negative beta are possible for investments that tend to go down when the market goes up, and vice versa.

# Summary

From the Gold Dataset , I conclude the following Observation:
    1. Pred is partially filled
    2. New is filled.
    3. Pred is Polynomially related to the inputs with degree 3 . 
    4. New is Linearly related to the inputs
    5. Build up the model with 0.9999 r2_score.
    
From the Stock(Ashoka) and Nifty Dataset , I conclude the following Observation:
    1. Beta value for Daily Return is 1.60861623
    2. Beta Value for Monthly Return is 1.37515766